resources/views/dormitory/rooms/app.blade.php

```diff
@@ -145,46 +153,46 @@
+                        <path id="room225" style="fill: {{getRoomColor(225, $roomColors)}}" data-info="{{ getRoomResidents(225, $rooms) }}" class="st1" d="M113.2,586.1l-0.6,55.8l54.4-0.2l0.4-69.6H145V586L113.2,586.1z"/>
                         <path id="em2fiufurdo" class="st2" d="M173.8,282.3l0.6,63.4l14.4-0.4l-0.4-15.5l3.6-0.2v16.1h72.2l-0.2-63.6h-72.6l0.6,34.8
                             l-3.2-0.6l-0.2-34.6L173.8,282.3z"/>
                         <path id="em2fiuwc" class="st2" d="M174.2,350.2l0.6,46.4h30.7v-6h3.6l-0.2,6l55.6-0.2l0.2-46.2l-56.3,0.2l0.4,28.3l-3.2-0.6
                             l-0.4-27.7L174.2,350.2z"/>
-                        <path id="room224" style="fill: {{$roomColors->where('id',224)->pluck('color')->join(',')}}" data-info="{{ $users->where('room', 224)->pluck('name')->join(';') }}" class="st1" d="M174.9,525.5h51.4l-0.2,92l-51.4,0.4L174.9,525.5z"/>
-                        <path id="room223" style="fill: {{$roomColors->where('id',223)->pluck('color')->join(',')}}" data-info="{{ $users->where('room', 223)->pluck('name')->join(';') }}" class="st1" d="M228.6,525.5l43.4,0.4l0.2,91.9h-44.2v-92.2H228.6z"/>
-                        <path id="room222" style="fill: {{$roomColors->where('id',222)->pluck('color')->join(',')}}" data-info="{{ $users->where('room', 222)->pluck('name')->join(';') }}" class="st1" d="M274.5,525.5l47.9,0.2l0.6,91.9l-48.5,0.2L274.5,525.5z"/>
-                        <path id="room221" style="fill: {{$roomColors->where('id',221)->pluck('color')->join(',')}}" data-info="{{ $users->where('room', 221)->pluck('name')->join(';') }}" class="st1" d="M324.4,525.5h46.4v92l-46.4,0.2V525.5z"/>
-                        <path id="room220" style="fill: {{$roomColors->where('id',220)->pluck('color')->join(',')}}" data-info="{{ $users->where('room', 220)->pluck('name')->join(';') }}" class="st1" d="M372.5,525.5l51.1,0.2l-0.4,92h-50.5L372.5,525.5z"/>
+                        <path id="room224" style="fill: {{getRoomColor(224, $roomColors)}}" data-info="{{ getRoomResidents(224, $rooms) }}" class="st1" d="M174.9,525.5h51.4l-0.2,92l-51.4,0.4L174.9,525.5z"/>
+                        <path id="room223" style="fill: {{getRoomColor(223, $roomColors)}}" data-info="{{ getRoomResidents(223, $rooms) }}" class="st1" d="M228.6,525.5l43.4,0.4l0.2,91.9h-44.2v-92.2H228.6z"/>
+                        <path id="room222" style="fill: {{getRoomColor(222, $roomColors)}}" data-info="{{ getRoomResidents(222, $rooms) }}" class="st1" d="M274.5,525.5l47.9,0.2l0.6,91.9l-48.5,0.2L274.5,525.5z"/>
+                        <path id="room221" style="fill: {{getRoomColor(221, $roomColors)}}" data-info="{{ getRoomResidents(221, $rooms) }}" class="st1" d="M324.4,525.5h46.4v92l-46.4,0.2V525.5z"/>
+                        <path id="room220" style="fill: {{getRoomColor(220, $roomColors)}}" data-info="{{ getRoomResidents(220, $rooms) }}" class="st1" d="M372.5,525.5l51.1,0.2l-0.4,92h-50.5L372.5,525.5z"/>
```

Comment:
Instead of the style, use the materialize colour classes (https://materializecss.github.io/materialize/color.html), also you can get it from the model.
For the coordinates, paste them into a new file (like we did with the countries), and you can also load them in the controller.
This way, you can make this in a foreach.

Reply:
Sadly I need the `fill ` CSS property to change not the `background-color`, so I couldn't use the materialize classes. But I tried my best to make the code readable in this regard.